src/migrator.rs

```diff
@@ -88,8 +88,8 @@ macro_rules! fn_migrator {
 
     ($migrate:tt, $name:tt, $builder:tt, $src_env:ty, $dst_env:ty) => {
         /// Same as the other migration methods, but automatically attempts to open the source
-        /// environment, ignores it if it doesn't exist or if it's empty, and finally attempts to
-        /// delete all of its supporting files.
+        /// environment, ignores it if it's invalid, if it doesn't exist or if it's empty.
```

Comment:
Nit: s/`..., ignores it if it's invalid, if it doesn't exist or if it's empty.`/`..., ignores the migration if the source environment is invalid, inexistent, or empty.`/

Since this migrator will delete all the supporting files, silently ignoring errors without notifying the caller might sound surprising to them. Let's add a note to the migration document so that the consumers can decide which migrator is the best suit for them.

Reply:
Note: it won't delete the files if they're invalid (corrupted, or unrelated). This is intentional.

I'll expand on picking the right migrator in the comments.